### PR TITLE
fix(indexer): rescue transient-stranded withdrawals before sender handoff (issue-262)

### PR DESCRIPTION
**Pull request by Huzaifa696**
## Summary

This PR fixes a issue 262 where withdrawals could become permanently stranded after a transient database or RPC error during transaction preparation. Before a withdrawal was handed to the sender, temporary failures while loading mint metadata left the row in `Processing` even though no transaction had been built or broadcast. Since the fetcher only processes `Pending` rows, these withdrawals were never retried automatically and eventually ended up in `ManualReview`. A brief outage could therefore strand an entire batch of withdrawals.

The fix rolls these withdrawals back from `Processing` to `Pending` whenever a transient error occurs **before** the sender is reached, allowing them to be retried after restart. This reuses the existing requeue mechanism already used by the sender for pre-broadcast failures, avoids any schema changes, and is guarded so it only runs when no transaction could have been sent. A retry limit also prevents endless requeue loops by moving repeatedly failing withdrawals to `ManualReview`, while transient database reads now benefit from the existing retry/backoff helper before being treated as failures.

The change is limited to the withdrawal preparation path and leaves deposits unchanged. Comprehensive tests verify successful requeueing, buffered-withdrawal handling, retry-limit behavior, rotation edge cases, end-to-end recovery after restart, database backoff, and that the normal successful flow is unaffected.
### Coverage Report

| Component | Lines Hit | Lines Total | Coverage | Artifact |
|-----------|-----------|-------------|----------|----------|
| Core | 11,846 | 13,489 | 87.8% | [rust-unit-coverage-reports](https://github.com/solana-foundation/solana-private-channels/actions/runs/29826702906) |
| Indexer | 29,719 | 33,035 | 90.0% | [rust-unit-coverage-reports](https://github.com/solana-foundation/solana-private-channels/actions/runs/29826702906) |
| Gateway | 1,325 | 1,515 | 87.5% | [rust-unit-coverage-reports](https://github.com/solana-foundation/solana-private-channels/actions/runs/29826702906) |
| Auth | 785 | 903 | 86.9% | [rust-unit-coverage-reports](https://github.com/solana-foundation/solana-private-channels/actions/runs/29826702906) |
| Withdraw Program | - | - | - | - |
| Escrow Program | - | - | - | - |
| DvP Swap Program | - | - | - | - |
| E2E Integration | 13,085 | 16,579 | 78.9% | [e2e-coverage-reports](https://github.com/solana-foundation/solana-private-channels/actions/runs/29826702906) |
| **Total** | **56,760** | **65,521** | **86.6%** | |

> Last updated: 2026-07-21 12:32:05 UTC by E2E Integration

### CODE REVIEW

File: indexer/src/operator/processor.rs [hunk 1]
```diff
@@ -252,6 +254,82 @@ async fn halt_withdrawal_pipeline(
     }
 }
 
+/// CAS one owned, unsent withdrawal `Processing -> Pending` so the fetcher can
+/// re-claim it. Safe because a transient error before any send proves nothing
+/// was broadcast and no signature was recorded. Mirrors the sender's
+/// pre-broadcast requeue: `Ok(false)` (row moved) and `Err` (write failed) are
+/// logged and the row is left `Processing` for recovery to reconcile.
```
[the remark this answers is not in the document]
Comment (Huzaifa696) [reply indexer/src/operator/processor.rs]: fixed

**Review by sebatustra — APPROVED** [review]
